### PR TITLE
Coverity fixes for net samples and kernel tests

### PR DESCRIPTION
**Pull request by lpereira**
Some last-minute fixes for minor issues in kernel tests and network samples.

### CODE REVIEW

**Review by jukkar — APPROVED** [review]
Whitespace issues in the code but other than that LGTM.

**Review by tbursztyka — APPROVED** [review]

**Comment by lpereira** [comment]
Rebased and repushed without the whitespace issues

**Review by nashif — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@lpereira Are there GitHub issues corresponding to the Coverity-IDs? If so, will you please reference them here?

**Review by tyrantlizard77 — APPROVED** [review]

File: samples/net/sockets/dumb_http_server/src/socket_dumb_http.c [hunk 1]
```diff
@@ -82,9 +82,19 @@ int main(void)
 		 * connection reset error).
 		 */
 		while (1) {
+			ssize_t r;
 			char c;
 
-			recv(client, &c, 1, 0);
+			r = recv(client, &c, 1, 0);
```
[the remark this answers is not in the document]
Comment (jukkar) [reply samples/net/sockets/dumb_http_server/src/socket_dumb_http.c]: IMHO it makes sense to always check return codes even for small samples. It is a good habit and if someone takes code snippet to some other program, the error handling example is there already.

**Review by MaureenHelm — APPROVED** [review]
Fixes #6093 
Fixes #8063 
Fixes #7716

**Pull request merged**